File: InsuranceCompany.Core/Models/InsuranceTypeSurvey.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace InsuranceCompany.Core;

public partial class InsuranceTypeSurvey
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    [ForeignKey(nameof(InsuranceSurvey))]
    public Guid? InsuranceSurveyId { get; set; }

    [ForeignKey(nameof(InsuranceRate))]
    public Guid? InsuranceRateId { get; set; }

    public virtual InsuranceSurvey? InsuranceSurvey { get; set; }

    public virtual InsuranceRate? InsuranceRate { get; set; }
}

File: InsuranceCompany.Client/src/app/models/InsuranceTypeSurvey.ts
import { InsuranceSurvey } from "./InsuranceSurvey";

export class InsuranceTypeSurvey {
    id : string;
    insuranceSurveyId : string;
    insuranceRateId : string;
    insuranceSurvey : InsuranceSurvey;

    constructor(id: string, insuranceSurveyId : string,
    insuranceRateId : string,insuranceSurvey : InsuranceSurvey) {
        
        this.id = id;
        this.insuranceSurveyId = insuranceSurveyId;
        this.insuranceRateId = insuranceRateId;
        this.insuranceSurvey = insuranceSurvey;
    }
}